src/timeline/Timeline.js

```diff
@@ -0,0 +1,216 @@
+  }
+
+  // _renderRedLine() {
+  //   const offset = 100;
+  //   const {start = 0} = this.props;
+  //   const timeNowHour = moment().hour();
+  //   const timeNowMin = moment().minutes();
+  //   return (
+  //     <View
+  //       key={'timeNow'}
+  //       style={[
+  //         this.styles.lineNow,
+  //         {
+  //           top: offset * (timeNowHour - start) + (offset * timeNowMin) / 60,
+  //           width: dimensionWidth - 20,
+  //         },
+  //       ]}
+  //     />
+  //   );
+  // }
+
+  _renderLines() {
+    const {format24h, start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+
+    return range(start, end + 1).map((i, index) => {
+      let timeText;
+      if (i === start) {
+        timeText = '';
+      } else if (i < 12) {
+        timeText = !format24h ? `${i} AM` : i;
+      } else if (i === 12) {
+        timeText = !format24h ? `${i} PM` : i;
+      } else if (i === 24) {
+        timeText = !format24h ? '12 AM' : 0;
+      } else {
+        timeText = !format24h ? `${i - 12} PM` : i;
+      }
+      return [
+        <Text
+          key={`timeLabel${i}`}
+          style={[this.styles.timeLabel, {top: offset * index - 6}]}>
+          {timeText}
+        </Text>,
+        i === start ? null : (
+          <View
+            key={`line${i}`}
+            style={[
+              this.styles.line,
+              {top: offset * index, width: dimensionWidth - 20},
+            ]}
+          />
+        ),
+        <View
+          key={`lineHalf${i}`}
+          style={[
+            this.styles.line,
+            {top: offset * (index + 0.5), width: dimensionWidth - 20},
+          ]}
+        />,
+      ];
+    });
+  }
+
+  _renderTimeLabels() {
+    const {start = 0, end = 24} = this.props;
+    const offset = this.calendarHeight / (end - start);
+    return range(start, end).map((item, i) => {
+      return (
+        <View key={`line${i}`} style={[this.styles.line, {top: offset * i}]} />
+      );
+    });
+  }
+
+  _onEventTapped(event) {
+    this.props.eventTapped(event);
+  }
+
+  _renderEvents() {
+    const {packedEvents} = this.state;
+    let events = packedEvents.map((event, i) => {
+      const style = {
+        left: event.left,
+        height: event.height,
+        width: event.width,
+        top: event.top,
+      };
+
+      const eventColor = {
+        backgroundColor: event.color,
+      };
+
+      // Fixing the number of lines for the event title makes this calculation easier.
+      // However it would make sense to overflow the title to a new line if needed
+      const numberOfLines = Math.floor(event.height / TEXT_LINE_HEIGHT);
+      const formatTime = this.props.format24h ? 'HH:mm' : 'hh:mm A';
+      return (
+        <TouchableOpacity
+          activeOpacity={0.5}
+          onPress={() => this._onEventTapped(this.props.events[event.index])}
+          key={i}
+          style={[this.styles.event, style, event.color && eventColor]}>
```

Comment:
why checking `event.color &&...`? You can pass `{backgroundColor: undefined}`

Reply:
@Inbal-Tish Done!